src/database.py

```diff
@@ -0,0 +1,84 @@
+# Copyright 2024 Canonical Ltd.
+# See LICENSE file for licensing details.
+
+"""Provide the DatabaseObserver class to handle database relation and state."""
+
+import typing
+
+from charms.data_platform_libs.v0.data_interfaces import DatabaseRequires
```

Comment:
Lib is missing in this PR, should be part of the files added

Reply:
True, added.